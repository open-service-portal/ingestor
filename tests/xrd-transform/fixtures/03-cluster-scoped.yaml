apiVersion: apiextensions.crossplane.io/v2
kind: CompositeResourceDefinition
metadata:
  name: clusters.infrastructure.com
  annotations:
    backstage.io/description: "Cluster-scoped infrastructure provisioning"
    openportal.dev/tags: "infrastructure,cluster"
spec:
  group: infrastructure.com
  names:
    kind: Cluster
    plural: clusters
  scope: Cluster
  versions:
    - name: v1alpha1
      served: true
      referenceable: true
      schema:
        openAPIV3Schema:
          type: object
          properties:
            spec:
              type: object
              properties:
                region:
                  type: string
                  description: Cloud region
                  default: us-west-2
                nodeCount:
                  type: integer
                  description: Number of nodes
                  minimum: 1
                  maximum: 100
                  default: 3
              required:
                - region
