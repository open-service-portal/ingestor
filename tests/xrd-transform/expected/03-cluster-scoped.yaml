apiVersion: scaffolder.backstage.io/v1beta3
kind: Template
metadata:
  name: clusters-infrastructure-com
  title: Cluster
  description: Create and manage Cluster resources
  tags:
    - crossplane
    - infrastructure-com
    - infrastructure
    - cluster
  annotations:
    backstage.io/managed-by: xrd-transform
    backstage.io/managed-by-location: provider:XRDTemplateEntityProvider
    backstage.io/managed-by-origin-location: provider:XRDTemplateEntityProvider
    crossplane.io/xrd-name: clusters.infrastructure.com
    crossplane.io/xrd-group: infrastructure.com
spec:
  owner: platform-team
  type: crossplane-resource
  parameters:
    - title: Resource Configuration
      required:
        - name
        - region
      properties:
        name:
          title: Name
          type: string
          description: Name of the Cluster
          pattern: ^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
        region:
          title: Region
          type: string
          description: Cloud region
          default: null
        nodeCount:
          title: NodeCount
          type: number
          description: Number of nodes
          default: 3
          minimum: 1
          maximum: 100
  steps:
    - id: workflow-info
      name: Direct Apply Workflow Information
      action: debug:log
      input:
        message: |
          ‚ö° This template uses direct apply workflow:

          1. Creates your Cluster resource immediately in the cluster
          2. Registers the resource in the Backstage catalog

          Your resource will be created as soon as you click Create.
      output:
        text:
          - title: ‚ö° Direct Apply Workflow
            content: >
              This template follows a **direct apply workflow** for immediate
              resource creation:


              **What happens when you click Create:**


              1. **Create Resource Immediately** üöÄ
                 - Your Cluster resource is created directly in the Kubernetes cluster
                 - No approval required - resource is active immediately
                 - Applied to cluster: `rancher-desktop`

              2. **Register in Catalog** üìù
                 - The resource is automatically registered in the Backstage software catalog
                 - You can track and manage it through Backstage

              **‚úÖ Important**: Your resource is **created immediately** without
              approval workflow.
    - id: create-resource
      name: Create Cluster
      action: kube:apply
      input:
        manifest: |
          apiVersion: infrastructure.com/v1alpha1
          kind: Cluster
          metadata:
            name: ${{ parameters.name }}
          spec: ${{ parameters }}
        namespaced: false
      output:
        text:
          - title: Resource Details
            content: |
              **Resource Created Successfully!** ‚úÖ

              **Resource Information:**
              - **Name**: `$`
              - **Type**: Cluster
              - **API Version**: infrastructure.com/v1alpha1
              - **Cluster**: rancher-desktop
              - **Scope**: Cluster-wide

              Your resource is now active in the cluster and ready to use.
    - id: register
      name: Register in Catalog
      action: catalog:register
      input:
        repoContentsUrl: ${{ steps.create-resource.output.repoContentsUrl }}
        catalogInfoPath: /catalog-info.yaml
      output:
        links:
          - title: View in Catalog
            icon: catalog
            url: ${{ steps['register'].output.entityRef }}
        text:
          - title: Catalog Registration
            content: |
              **Registered in Backstage Catalog** üìù

              Your Cluster resource has been added to the software catalog.
              You can now track, monitor, and manage it through Backstage.

---
apiVersion: backstage.io/v1alpha1
kind: API
metadata:
  name: clusters-infrastructure-com-api
  title: Cluster API
  description: API specification for Cluster
  tags:
    - crossplane
    - ingestor
    - infrastructure-com
  annotations:
    backstage.io/managed-by: xrd-transform
    backstage.io/managed-by-location: provider:XRDTemplateEntityProvider
    backstage.io/managed-by-origin-location: provider:XRDTemplateEntityProvider
    crossplane.io/xrd-name: clusters.infrastructure.com
spec:
  type: openapi
  lifecycle: experimental
  owner: platform-team
  definition: |
    openapi: 3.0.0
    info:
      title: Cluster API
      version: v1alpha1
      description: OpenAPI specification for Cluster
    paths:
      /clusters:
        get:
          summary: List clusters
          responses:
            '200':
              description: Success
        post:
          summary: Create Cluster
          requestBody:
            content:
              application/json:
                schema:
                  type: object
                  properties:
                    spec:
                      type: object
                      description: Cluster configuration
                      properties:
                        region:
                          type: string
                          description: "Cloud region"
                        nodeCount:
                          type: number
                          description: "Number of nodes"
                          minimum: 1
                          maximum: 100
                      required:
                        - region
                    status:
                      type: object
                      properties:
                        ready:
                          type: boolean
                          description: Whether the resource is ready
                        phase:
                          type: string
                          description: Current phase of the resource
                        message:
                          type: string
                          description: Status message
          responses:
            '201':
              description: Created

